.github/workflows/binder-on-pr.yml

```diff
@@ -4,6 +4,10 @@ on:
   pull_request_target:
     types: [opened]
 
+permissions:
+  pull-requests:
+    write
```

Comment:
This raises all sorts of red flags in my head about allowing arbitrary code execution, or pushing code to the repo, etc. Is there somewhere an overview of the security issues involved?

Reply:
There is an issue about this in the `jupyterlab` team compass: https://github.com/jupyterlab/team-compass/issues/125

With links to PRs implementing the changes.

Reply:
I gained some experience with this over in the closing answered issue PR. I now think it's fine.